### ISSUE TITLE
Accessor returns value other than the one that was set

### ISSUE THREAD

**Issue by kjsanger** [wtsi-npg/npg_id_generation#19]
When `plate_number` is set to `1`, the attribute becomes `None`. The attribute should still be `1`.

```
from npg_id_generation.pac_bio import PacBioEntity
x = PacBioEntity(run_name="run1", well_label="A1", plate_number=1, tags="AAA")
x.plate_number
x.plate_number == 1
False
```

If `plate_number` is set to `2` or greater, then the attribute remains as set. It's fine for the generated checksum to use the plate number `1` in a special, backwards-compatible way, but that should not be exposed by silently mutating the attribute.

**Comment by nerdstrike** [comment]
I believe this is intended behaviour, because of all the older Revio runs from before multi-plate that don't set `plate_number=1`. We would have had to rebuild their product IDs in the databases. I was in favour of patching, but the collective decision was to protect the older data from unintended consequences.

**Comment by kjsanger** [comment]
I understand why it's beening done, however it's still wrong for an attribute to be changed from what it's explicitly set to. That's what I mean by the implementation of the ID generation not silently mutating the attribute as a side-effect of which data it includes.

**Comment by kjsanger** [comment]
Fixed by #22
